metaflow/metadata/metadata.py

```diff
@@ -364,7 +383,52 @@ def get_object(cls, obj_type, sub_type, filters, *args):
         if sub_type == 'metadata' and obj_type != 'task':
             raise MetaflowInternalError(msg='Metadata can only be retrieved at the task level')
 
-        return cls._get_object_internal(obj_type, type_order, sub_type, sub_order, filters, *args)
+        if attempt is not None:
+            try:
+                attempt_int = int(attempt)
+                if attempt_int < 0:
+                    raise ValueError("Attempt can only be positive")
+            except ValueError:
+                raise ValueError("Attempt can only be a positive integer")
+        else:
+            attempt_int = None
+
+        pre_filter = cls._get_object_internal(
+            obj_type, type_order, sub_type, sub_order, filters, attempt_int, *args)
+        if attempt_int is None or sub_order != 6:
+            # If no attempt or not for metadata, just return as is
+            return pre_filter
+        # Otherwise, we filter out the metadata. If we have attempt_id information
+        # (post MF 2.4.0), it's easy; if not, we do our best to reconstruct
+        # the order.
```

Comment:
you could move this special logic to handle older versions in a function of its own. Now it adds a lot of complexity in `get_object`

Reply:
ok, will refactor.

Reply:
so to clarify, just this part of the function that builds the metadata filtered out when we don't have the attempt value right?